components/core/src/clp_s/clp-s.cpp

```diff
@@ -271,10 +271,11 @@ int main(int argc, char const* argv[]) {
 
         clp_s::JsonConstructorOption option;
         option.output_dir = command_line_arguments.get_output_dir();
+        option.archives_dir = std::filesystem::path{archives_dir};
```

Comment:
Why turn this into a filesystem path? `option.archives_dir` is a string. Did you want to turn `option.archives_dir` and the parameter to `ArchiveReader::open` into a filesystem path?